### ISSUE TITLE
Rename symbol - No file available to rename

### ISSUE THREAD

**Issue by tekumara** [rust-lang/rust-analyzer#14379]
When trying to rename a symbol (by pressing F2) I get:
```
[Error - 20:35:06] Request textDocument/rename failed.
  Message: No file available to rename
  Code: -32602 
```

https://user-images.githubusercontent.com/125105/226166521-2de28be0-4d31-4f5b-83bc-9dc221efa561.mp4

**rust-analyzer version**: rust-analyzer version: 0.3.1435-standalone (f1e51afa4 2023-03-12)

**rustc version**: rustc 1.68.0 (2c8cc3432 2023-03-06) (built from a source tarball)

**Comment by lnicola** [comment]
Probably a duplicate of #13388.

**Comment by djc** [comment]
I also noticed this today. Feels likely that there is a regression somehow? The code where I'm running into this is using `#[tracing::instrument]`.

**Comment by HKalbasi** [comment]
Can someone check if it happens before #14232 ? (Or more generally bisect it)

**Comment by Veykril** [comment]
It is likely that PR + the one where I fixed reference search not working in attributes again. 
https://github.com/veykril/rust-analyzer/blob/66636939a63237fb7a6d1198dd9f92514807621e/crates/ide-db/src/rename.rs#L466

This line should try mapping out of macro calls and only error if the definition can't be mapped to macro inputs (we should also adjust the error stating that the definition comes from a macro and hence can't be renamed)

**Comment by mikkelens** [comment]
I have also encountered this error and I didn't manage to get symbol renaming working again. It worked earlier today, even for a while after reloading VSCode with the newest edition of rust-analyzer. No idea what prompted it to break only after a while.

**Comment by memark** [comment]
Here is an MRE:

Renaming the variable _doesn't_ work
```rust
#[tokio::main]
async fn main() {
    let a = 1;
}
```

Renaming the variable _does_ work
```rust
async fn main() {
    let a = 1;
}
```

**Comment by ruabmbua** [comment]
Also started to experience this with `#[async_std::main] `

**Comment by memark** [comment]
Also noted it has other effects, e.g. "Extract into function" refactoring disappears from the right-click context menu in VS Code.

**Comment by ThinkerDreamer** [comment]
I just had the same issue with a slightly different error message while using the `#[axum_macros::debug_handler]` macro attribute.
```
[Error - 2:49:13 PM] Request textDocument/rename failed.
  Message: Cannot rename a non-local definition
  Code: -32602 
  ```
  rust-analyzer version: 3.1924
  rustc version: 1.74.0

**Comment by Veykril** [comment]
That will need more information than that, ideally a snippet that caused this

**Comment by ThinkerDreamer** [comment]
```
#[axum_macros::debug_handler]
pub async fn login(Json(input): Json<LoginRequest>) -> impl IntoResponse {
    println!("{input:?}");
    StatusCode::OK.into_response()
}
```
Trying to rename the `login` fn doesn't work when I have the macro uncommented.
